### PR TITLE
FT: Set one pair of accessKey/secretKey with env variables

### PR DESCRIPTION
**Pull request by nicolas2bert**

### CODE REVIEW

File: lib/auth/in_memory/builder.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/** build simple authdata with only one account
+ * @param {string} accessKey - account's accessKey
+ * @param {string} secretKey - account's secretKey
+ * @return {object} authdata - authdata with account's accessKey and secretKey
+ */
+export function buildAuthDataAccount(accessKey, secretKey) {
+    return {
+        accounts: [{
+            name: 'CustomAccount',
+            email: 'customaccount1@sampling.com',
```
Comment (LaurenSpiegel) [inline lib/auth/in_memory/builder.js]: maybe call it 'customaccount1@setbyenv.com'

File: lib/Config.js [hunk 2]
```diff
@@ -435,7 +436,11 @@ class Config {
             if (process.env.S3AUTH_CONFIG) {
                 authfile = process.env.S3AUTH_CONFIG;
             }
-            const authData = require(authfile);
+            let authData = require(authfile);
+            if (process.env.ACCESS_KEY && process.env.SECRET_KEY) {
```
Comment (rahulreddy) [inline lib/Config.js]: Suggestion, why not use `SCALITY_ACCESS_KEY_ID` and `SCALITY_SECRET_ACCESS_KEY` which would be similar to `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` used by AWS. When we have multiple backends we can use both sets to work with AWS/SCALITY.

Comment (nicolas2bert) [inline lib/Config.js]: Makes sense! @rahulreddy

File: DOCKER.md [hunk 3]
```diff
@@ -25,19 +25,19 @@ you and your users can direct s3 server requests to example.com.
...
+#### SCALITY_ACCESS_KEY_ID and SCALITY_SECRET_ACCESS_KEY
 
 These variables specify authentication credentials for an account
-named "Docker".
+named "CustomAccount".
 
 You can set credentials for many accounts by editing `conf/authdata.json`
 (see below for further info),
 but if you just want to specify one set of your own,
 you can use these environment variables.
```
Comment (LaurenSpiegel) [inline DOCKER.md]: Please add here as well -- Note that if you set these environment variables, the conf/authdata.json will be ignored.

File: README.md [hunk 4]
```diff
@@ -71,6 +71,22 @@ This starts an S3 server on port 8000.
 The default access key is accessKey1 with
 a secret key of verySecretKey1.
 
+## Run it with environment variables
```
Comment (LaurenSpiegel) [inline README.md]: I would do this as a separate section (not a run it) with a heading such as "Setting your own access key and secret key pairs"

Then I would say there are 2 options for setting your own, the env variables or the authdata.json file.

File: DOCKER.md [hunk 5]
```diff
@@ -25,19 +25,19 @@ you and your users can direct s3 server requests to example.com.
 docker run -d --name s3server -p 8000:8000 -e HOST_NAME=new.host.com scality/s3server
 ```
 
-#### ACCESS_KEY and SECRET_KEY
```
Comment (rahulreddy) [inline DOCKER.md]: You can add a note at the bottom that the old `ACCESS_KEY` and `SECRET_KEY` environment variables are now deprecated.

Comment (nicolas2bert) [inline DOCKER.md]: Good idea - Done :)

File: DOCKER.md (hunk 3 above)
Comment (nicolas2bert) [inline DOCKER.md]: Done :)

**Comment by nicolas2bert** [comment]
@ironman-machine

**Comment by nicolas2bert** [comment]
E2E passed http://ci.ironmann.io/gh/scality/Integration/10174#build-parameters

File: README.md [hunk 6]
```diff
@@ -71,6 +71,23 @@ This starts an S3 server on port 8000.
 The default access key is accessKey1 with
 a secret key of verySecretKey1.
 
+## Setting your own access key and secret key pairs
+
+You can set credentials for many accounts by editing `conf/authdata.json` but if
+you want to specify one set of your own credential, you can use
```
Comment (bennettbuchanan) [inline README.md]: ~credential~ credentials

File: lib/Config.js [hunk 7]
```diff
@@ -469,7 +470,13 @@ class Config {
             if (process.env.S3AUTH_CONFIG) {
                 authfile = process.env.S3AUTH_CONFIG;
             }
-            const authData = require(authfile);
+            let authData = require(authfile);
```
Comment (bennettbuchanan) [inline lib/Config.js]: If `authfile` is invalid JSON, this would throw an error. Maybe it's okay, but it could be helpful to check that first.

Comment (nicolas2bert) [inline lib/Config.js]: I guess we are ok throwing an error if `authfile` is invalid JSON, @LaurenSpiegel can confirm.

Comment (LaurenSpiegel) [inline lib/Config.js]: yes, let them know early that there is a problem

**Pull request merged**